scio-test/src/test/scala/com/spotify/scio/coders/CoderTest.scala

```diff
@@ -43,161 +50,63 @@ import java.time.{Instant, LocalDate}
-  implicit val caseClassWithExplicitCoderCoder: Coder[CaseClassWithExplicitCoder] =
-    Coder.beam(new AtomicCoder[CaseClassWithExplicitCoder] {
-      val sc = StringUtf8Coder.of()
-      val ic = VarIntCoder.of()
-      def encode(value: CaseClassWithExplicitCoder, os: OutputStream): Unit = {
-        ic.encode(value.i, os)
-        sc.encode(value.s, os)
-      }
-      def decode(is: InputStream): CaseClassWithExplicitCoder = {
-        val i = ic.decode(is)
-        val s = sc.decode(is)
-        CaseClassWithExplicitCoder(i, s)
-      }
-    })
-}
-
-// nested
-case class NestedB(x: Int)
-case class NestedA(nb: NestedB)
-
-// recursive
-case class SampleField(name: String, fieldType: SampleFieldType)
-sealed trait SampleFieldType
-case object IntegerType extends SampleFieldType
-case object StringType extends SampleFieldType
-case class RecordType(fields: List[SampleField]) extends SampleFieldType
-case class Recursive(a: Int, rec: Option[Recursive] = None)
-
-// private
-class PrivateClass private (val value: Long) extends AnyVal
-object PrivateClass {
-  def apply(l: Long): PrivateClass = new PrivateClass(l)
-}
-case class UsesPrivateClass(privateClass: PrivateClass)
-
-// proto
-case class ClassWithProtoEnum(s: String, `enum`: OuterClassForProto.EnumExample)
-
-// serial UID
-@SerialVersionUID(1)
-sealed trait TraitWithAnnotation
-@SerialVersionUID(2)
-final case class FirstImplementationWithAnnotation(s: String) extends TraitWithAnnotation
-@SerialVersionUID(3)
-final case class SecondImplementationWithAnnotation(i: Int) extends TraitWithAnnotation
-
-// AnyVal
-final case class AnyValExample(value: String) extends AnyVal
-
-// Non deterministic
-final case class NonDeterministic(a: Double, b: Double)
-
-class ClassWrapper() {
-  case class InnerCaseClass(str: String)
-
-  def runWithImplicit(implicit
-    c: Coder[InnerCaseClass]
-  ): Unit =
-    InnerCaseClass("51") coderShould roundtrip()
-
-  def run(): Unit =
-    InnerCaseClass("51") coderShould roundtrip()
-}
-
-object TopLevelObject {
-  case class InnerCaseClass(str: String)
-}
-
 final class CoderTest extends AnyFlatSpec with Matchers {
 
   val userId: UserId = UserId(Seq[Byte](1, 2, 3, 4))
   val user: User = User(userId, "johndoe", "johndoe@spotify.com")
 
-  /*
-   * Case class nested inside another class. Do not move outside
-   * */
-  case class InnerCaseClass(str: String)
-
-  /*
-   * Object nested inside another class. Do not move outside
-   * */
-  object InnerObject {
-    case class InnerCaseClass(str: String)
-  }
-
   def materialize[T](coder: Coder[T]): BCoder[T] =
     CoderMaterializer.beam(PipelineOptionsFactory.create(), coder)
 
-  "Coders" should "support primitives" in {
-    1 coderShould roundtrip()
-    'a' coderShould roundtrip()
-    "yolo" coderShould roundtrip()
-    4.5 coderShould roundtrip()
+  it should "support primitives" in {
+    false coderShould roundtripToBytes(Array(0)) and beOfType[Beam[_]] and
+      materializeTo[BooleanCoder]
```

Comment:
I think it makes sense to assert types of underlying coders to make sure implicits + pattern matching in `materialize` worked as expected

Reply:
That's damn good syntax!